File: C#/TopflytechCodec/TopflytechCodec/Entities/RS232Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TopflytechCodec.Entities
{
    /// <summary>
    /// The type RS232 message.Protocol number is 25 25 09.
    /// </summary>
    public class RS232Message:Message
    {
        private DateTime date;
        public System.DateTime Date
        {
            get { return date; }
            set { date = value; }
        }
        private bool isIgnition;
        /// <summary>
        /// Is vehicle ignition 
        /// </summary>
        public bool IsIgnition
        {
            get { return isIgnition; }
            set { isIgnition = value; }
        }

        private int rs232DataType;

        public int Rs232DataType
        {
            get { return rs232DataType; }
            set { rs232DataType = value; }
        }
        private List<Rs232DeviceMessage> rs232DeviceMessageList;

        internal List<Rs232DeviceMessage> Rs232DeviceMessageList
        {
            get { return rs232DeviceMessageList; }
            set { rs232DeviceMessageList = value; }
        }

        public static int OTHER_DEVICE_DATA = 0;
        public static int TIRE_DATA = 1;
        public static int RDID_DATA = 3;
        public static int FINGERPRINT_DATA = 2;
        public static int CAPACITOR_FUEL_DATA = 4;
        public static int ULTRASONIC_FUEL_DATA = 5;
    }
}

File: Java/topflytechCodec/src/com/topflytech/codec/entities/RS232Message.java
package com.topflytech.codec.entities;
 
import java.util.Date;
import java.util.List;

/**
 * The type RS232 message.Protocol number is 25 25 09.
 */
public class RS232Message extends Message{
    /**
     * Gets date.
     *
     * @return the date
     */
    public Date getDate() {
        return date;
    }

    /**
     * Sets date.
     *
     * @param date the date
     */
    public void setDate(Date date) {
        this.date = date;
    }

    /**
     * Is vehicle ignition boolean.
     *
     * @return the boolean
     */
    public boolean isIgnition() {
        return isIgnition;
    }

    /**
     * Sets ignition.
     *
     * @param ignition the ignition
     */
    public void setIgnition(boolean ignition) {
        this.isIgnition = ignition;
    }

    private Date date;
    private boolean isIgnition;


    public List<Rs232DeviceMessage> getRs232DeviceMessageList() {
        return rs232DeviceMessageList;
    }

    public void setRs232DeviceMessageList(List<Rs232DeviceMessage> rs232DeviceMessageList) {
        this.rs232DeviceMessageList = rs232DeviceMessageList;
    }

    public int getRs232DataType() {
        return rs232DataType;
    }

    public void setRs232DataType(int rs232DataType) {
        this.rs232DataType = rs232DataType;
    }

    private int rs232DataType;
    private List<Rs232DeviceMessage> rs232DeviceMessageList;

    public final static int OTHER_DEVICE_DATA = 0;
    public final static int TIRE_DATA = 1;
    public final static int RDID_DATA = 3;
    public final static int FINGERPRINT_DATA = 2;
    public final static int CAPACITOR_FUEL_DATA = 4;
    public final static int ULTRASONIC_FUEL_DATA = 5;

}
